sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -212,7 +225,16 @@ case class FileSourceScanExec(
     relation.fileFormat.vectorTypes(
       requiredSchema = requiredSchema,
       partitionSchema = relation.partitionSchema,
-      relation.sparkSession.sessionState.conf)
+      relation.sparkSession.sessionState.conf).map { vectorTypes =>
+        val metadataVectorClz =
+          if (relation.sparkSession.sessionState.conf.offHeapColumnVectorEnabled) {
+            classOf[OffHeapColumnVector].getName
+          } else {
+            classOf[OnHeapColumnVector].getName
```

Comment:
since we will change to use a constant vector soon, how about we always use `OnHeapColumnVector` for now,  to simplify the code?

Reply:
good idea! thanks!